### DISCUSSION TITLE
2 js files created on build when js script is preloaded

### DISCUSSION THREAD

**Discussion by eldk** [parcel-bundler/parcel#6936] [Q&A]
Hello,

I come back to parceljs from one project.

I was using Parcel V2 for another project, until last days of may.

I now notice some changes on build : 

#### before (until june) : 

```html
<head>
  <link rel="preload" href="./index.js" as="script">
</head>
<body>
  <script src="./index.js" type="module"></script>
</body>
```
was builded with only one generated js : index.tag.js

#### now : (using next parceljs - same with latest)

Two js files are builded and included in index.html
1 - the first one (preload) with the name of the app - `<link rel="preload" href="frontend.fef217f5.js" as="script">`
2 - the second one `<script src="index.b218514d.js" type="module">`

dist/frontend.fef217f5.js    123.05 KB    4.46s
dist/index.b218514d.js       102.28 KB    2.85s

If I remove `<link rel="preload" href="./index.js" as="script">`  only one js is created.

```json
"name": "frontend",
  "version": "0.0.1",
  "description": "",
  "source": "src/index.html",
  "scripts": {
    "dev": "parcel src/index.html",
    "build": "parcel build --public-url . src/index.html --no-source-maps",
    "prebuild": "rm -rf dist",
    "scrape": "node tests/scrape.test.js",
    "test": "node ./node_modules/mocha/bin/mocha --timeout=30000 tests"
  }, ...
```

Is it something wanted or am I missing something ?

Thanks,

Eric

**Comment by eldk** [comment]
With : `"build": "parcel build --public-url . src/index.html --no-source-maps --no-scope-hoist"` only one js file is created with the app name : `frontend.checksum.js`

**Comment by webadam1** [comment]
Hey!
I've also ran into parcel generating multiple js files after migrating from v1, but for me, the [_**differential bundling**_ part of the documentation](https://parceljs.org/features/targets/#differential-bundling) explained the behaviour.
I didn't have a `<link rel="preload" href="./index.js" as="script">` type deal in my code, so our cases might be different, but I'm leaving this comment in case someone else wondering about duplicate js files ends up here.
